lib/collection/src/hash_ring.rs

```diff
@@ -246,6 +248,54 @@ impl<T: Hash + Copy> Inner<T> {
             Inner::Fair { ring, .. } => ring.is_empty(),
         }
     }
+
+    pub fn len(&self) -> usize {
+        match self {
+            Inner::Raw(ring) => ring.len(),
+            Inner::Fair { ring, scale } => ring.len() / *scale as usize,
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Filter<T = ShardId> {
+    ring: Inner<T>,
+    filter: T,
+}
+
+impl<T> Filter<T> {
+    pub fn new(ring: Inner<T>, filter: T) -> Self {
+        Self { ring, filter }
+    }
+}
+
+impl<T> ReshardingCondition for Filter<T>
+where
+    T: fmt::Debug + Hash + PartialEq + Copy + 'static,
+{
+    fn check(&self, point_id: PointIdType) -> bool {
+        self.ring.get(&point_id) == Some(&self.filter)
+    }
+
+    fn estimate_cardinality(&self, points: usize) -> CardinalityEstimation {
+        CardinalityEstimation {
+            primary_clauses: vec![],
```

Comment:
Not sure about this one.

Reply:
Andrey said, `Resharding` condition does not need one. 🤷‍♀️

Or maybe we can benchmark this.

Reply:
I see your comment on it now. Thanks!